ForumProject/notifications/signals.py

```diff
@@ -0,0 +1,70 @@
+from django.db.models.signals import post_save, post_delete
+from django.dispatch import receiver
+from .models import Notification
+from projects.models import InvestorProject
+
+def create_notification(instance, follow_status_change=True):
+    '''
+    Helper function.
+    Create a new `Notification` record based on the `InvestorProject` instance.
+
+    This function creates a `Notification` record that represents a change in the 
+    relationship between an investor and a project. The trigger for the notification 
+    depends on whether it's a change in the following status or subscription share.
+
+    Parameters:
+    - instance (InvestorProject): The `InvestorProject` instance that triggered the notification.
+    - follow_status_change (bool): Whether the trigger is a change in the following status. 
+      Default is True. If False, the trigger is considered a change in the subscription share.
+    
+    Returns:
+    - Notification: The created `Notification` instance.
+    '''
+    Notification.objects.create(
+            project=instance.project,
+            startup=instance.project.startup,
+            investor=instance.investor,
+            trigger='follower(s) list changed' if follow_status_change else 'subscription changed',
+            initiator='investor',
+        )
```

Comment:
Consider adding validation to ensure all fields (`project`, `startup`, `investor`, etc.) are present and valid

Reply:
Done